iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/Replace3TwoWindingsTransformersByThreeWindingsTransformers.java

```diff
@@ -0,0 +1,541 @@
+                .setId(getId(twoR))
+                .setName(getName(twoR))
+                .setRatedU0(ratedU0);
+
+        addLeg(t3wAdder.newLeg1(), twoR.t2w1, isWellOrientedT2w1, ratedU0);
+        addLeg(t3wAdder.newLeg2(), twoR.t2w2, isWellOrientedT2w2, ratedU0);
+        addLeg(t3wAdder.newLeg3(), twoR.t2w3, isWellOrientedT2w3, ratedU0);
+        ThreeWindingsTransformer t3w = t3wAdder.add();
+
+        // t3w is not considered in regulatedTerminalControllers (created later in the model)
+        setLegData(t3w.getLeg1(), twoR.t2w1, isWellOrientedT2w1, regulatedTerminalControllers, twoR);
+        setLegData(t3w.getLeg2(), twoR.t2w2, isWellOrientedT2w2, regulatedTerminalControllers, twoR);
+        setLegData(t3w.getLeg3(), twoR.t2w3, isWellOrientedT2w3, regulatedTerminalControllers, twoR);
+
+        copyStarBusVoltageAndAngle(twoR.starBus, t3w);
+        List<PropertyR> lostProperties = new ArrayList<>();
+        lostProperties.addAll(copyProperties(twoR.t2w1, t3w));
+        lostProperties.addAll(copyProperties(twoR.t2w2, t3w));
+        lostProperties.addAll(copyProperties(twoR.t2w3, t3w));
+
+        List<ExtensionR> lostExtensions = copyExtensions(twoR, t3w);
+
+        // copy necessary data before removing
+        List<AliasR> t2wAliases = new ArrayList<>();
+        t2wAliases.addAll(getAliases(twoR.t2w1, "1", getEnd1(isWellOrientedT2w1)));
+        t2wAliases.addAll(getAliases(twoR.t2w2, "2", getEnd1(isWellOrientedT2w2)));
+        t2wAliases.addAll(getAliases(twoR.t2w3, "3", getEnd1(isWellOrientedT2w3)));
+
+        String t2w1Id = twoR.t2w1.getId();
+        String t2w2Id = twoR.t2w2.getId();
+        String t2w3Id = twoR.t2w3.getId();
+        String starVoltageId = twoR.starBus.getVoltageLevel().getId();
+        List<LimitsR> lostLimits = findLostLimits(twoR);
+
+        remove(twoR);
+
+        // after removing
+        List<AliasR> lostAliases = copyAliases(t2wAliases, t3w);
+
+        // warnings
+        if (!lostProperties.isEmpty()) {
+            lostProperties.forEach(propertyR -> logOrThrow(throwException, TWO_WINDINGS_TRANSFORMER + "'" + propertyR.t2wId + "' property '" + propertyR.propertyName + "' " + WAS_NOT_TRANSFERRED));
+        }
+        if (!lostExtensions.isEmpty()) {
+            lostExtensions.forEach(extensionR -> logOrThrow(throwException, TWO_WINDINGS_TRANSFORMER + "'" + extensionR.t2wId + "' extension '" + extensionR.extensionName + "' " + WAS_NOT_TRANSFERRED));
+        }
+        if (!lostAliases.isEmpty()) {
+            lostAliases.forEach(aliasR -> logOrThrow(throwException, TWO_WINDINGS_TRANSFORMER + "'" + aliasR.t2wId + "' alias '" + aliasR.alias + "' '" + aliasR.aliasType + "' " + WAS_NOT_TRANSFERRED));
+        }
+        if (!lostLimits.isEmpty()) {
+            lostLimits.forEach(limitsR -> logOrThrow(throwException, TWO_WINDINGS_TRANSFORMER + "'" + limitsR.t2wId + "' operationalLimitsGroup '" + limitsR.operationalLimitsGroupName + "' is lost"));
+        }
```

Comment:
Should we really throw exceptions here (if `throwException` is true)?
As stated in the comment, those are more warnings to raise than real problems, aren't they?

Reply:
Replaced by Log.warn